src/wallet/rpcdump.cpp

```diff
@@ -608,10 +628,18 @@ UniValue importwallet(const JSONRPCRequest& request)
             }
         }
         file.close();
-        pwallet->ShowProgress("", 100); // hide progress dialog in GUI
+        pwallet->ShowProgress("", 100, false, nullptr); // hide progress dialog in GUI
         pwallet->UpdateTimeFirstKey(nTimeBegin);
     }
-    pwallet->RescanFromTime(nTimeBegin, reserver, false /* update */);
+    pwallet->ShowProgress("", 100, false, nullptr); // hide progress dialog in GUI
+    int64_t scanned_time = pwallet->RescanFromTime(nTimeBegin, reserver, false /* update */);
+    if (scanned_time > nTimeBegin) {
+        if (pwallet->IsAbortingRescan()) {
```

Comment:
This is ugly, since RescanFromTime changes without a signature change. Better to have it throw an exception to be caught from callers, when cancelled.

Reply:
I don't quite understand what you mean.

Reply:
@luke-jr In what way does `RescanFromTime` change?

Reply:
`RescanFromTime` uses `ScanForWalletTransactions` internally. And the interface of `ScanForWalletTransactions` is changed in this pull request.

Reply:
That change was reverted.